include/data/field.h

```diff
@@ -36,6 +36,7 @@ struct FieldHighType {
     STRING,
     POINTER,
     ENUM,
+    MODE_GUARD,
```

Comment:
Simple enums won't allow it since the `GUARD` name wouldn't be unique (it would be possible with enum classes).